src/vs/workbench/services/workspaces/common/workspaceTrust.ts

```diff
@@ -239,19 +258,30 @@ export class WorkspaceTrustManagementService extends Disposable implements IWork
 			return true; // trust running tests with vscode-test
 		}
 
-		// Remote - remote authority explicitly sets workspace trust
-		if (this.environmentService.remoteAuthority && this._remoteAuthority?.options?.isTrusted !== undefined) {
+		// Remote - resolver explicitly sets workspace trust to TRUE
+		if (this.environmentService.remoteAuthority && this._remoteAuthority?.options?.isTrusted) {
 			return this._remoteAuthority.options.isTrusted;
 		}
 
-		if (this.workspaceService.getWorkbenchState() === WorkbenchState.EMPTY) {
-			// Use memento if present, otherwise default to restricted mode
-			// Workspace may transition to trusted based on the opened editors
-			return this._trustState.isTrusted ?? false;
+		// Canonical Uris not yet resolved
+		if (!this._canonicalUrisResolved) {
+			return false;
 		}
 
-		if (!this._initialized) {
-			return false;
+		// Empty workspace - use memento, open ediors, or user setting
+		if (this.workspaceService.getWorkbenchState() === WorkbenchState.EMPTY) {
+			// Use memento if present
+			if (this._trustState.isEmptyWorkspaceTrusted !== undefined) {
+				return this._trustState.isEmptyWorkspaceTrusted;
+			}
+
+			// Open editors
```

Comment:
this makes the flow a bit more confusing. i suggest renaming this property to something like `startupEditors` or `startupFiles`... because we only use this for initialization. unlike workspace uris, these should play any role in calculating workspace trust after initialization. I think the memento takes care of that here, but it is not clear from the naming or how they are populated above.